lib/control.rb

```diff
@@ -0,0 +1,112 @@
+class Control
```

Comment:
This class does feel heavy.  Could it be broken down into classes that handle it's separate responsibilities?